src/base-apis.js

```diff
@@ -277,7 +277,21 @@ MatrixBaseApis.prototype.loginWithPassword = function(user, password, callback)
     return this.login("m.login.password", {
         user: user,
         password: password,
-    }, callback);
+    }, (error, response) => {
+        if (response && response.access_token) {
+            this._http.opts.accessToken = response.access_token;
+        }
+
+        if (response && response.user_id) {
+            this.credentials = {
+                userId: response.user_id,
+            };
+        }
+
+        if(callback) {
```

Comment:
Might look nicer with promises/async await instead of a callback here:
```
... = async function(user, password, callback) {
  const response = await this.login(..., callback);
  if (response.access_token) {
   ...
  }
  if (response.user_id) {
   ...
  }
  return response;
}
```

Reply:
Hmm, not sure what is the right thing to do here actually. The promise is resolved before the user callback in `http-api.js:848`, but that will be run as a microtask. Right after, the user callback is called (synchronously). So the user callback code will run first, so setting the access token/user id in there is probably safest to make sure usages of the sdk using callbacks rather than promises don't break.